### DISCUSSION TITLE
Magic Forms v2

### DISCUSSION THREAD

**Discussion by skydiver** [skydiver/october-plugin-forms#231] [General]
> \* This post details current progress of Magic Forms v2 and will be updated regularly

## What's new
* new AJAX upload component using [Filepond](https://pqina.nl/filepond/)
* old upload component was removed (BREAKING CHANGE!)
* a few requested features (list pending)
* code cleanup & linting

## Pending / Completed list
- [x] create new upload component
- [x] server side validation: uploads file type
- [ ] server side validation: uploads size
- [ ] move some hardcoded strings to language file
- [ ] update docs with upload component instructions

## How to test
You can start using new version right now, but remember it can contain bugs or future breaking changes.

1. open your terminal and goto your OctoberCMS project
2. clone this repo using `git clone https://github.com/skydiver/october-plugin-forms.git plugins/martin/forms`
3. switch to `release/2.0.0` branch using `git checkout release/2.0.0`

### Documentation
Docs where update, screenshots re-taken and new features will be added.
Temporary url: https://mf-docs.vercel.app/ 

### Release date
No release date yet, development still in progress.

**Comment by cheuklam** [comment]
Thanks Skydiver for the update!
I got a new project that may need to use the plugin heavily, may I know if there is anything that I can help on the v2 version? as I might got some new feature need to add to the plugin to fulfill some of the requirement, e.g. an easier way for user to create 2 / 3 columns question field instead of over-riding the field col-d-12 to col-md-4 or col-md-6
